pkg/ext-proc/backend/pod_reconciler_test.go

```diff
@@ -145,22 +237,41 @@ func TestUpdateDatastore_PodReconciler(t *testing.T) {
 					},
 				},
 			},
-			wantPods: []string{basePod2.Name},
+			wantPods: []Pod{basePod2.Pod},
 		},
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			podReconciler := &PodReconciler{Datastore: test.datastore}
-			podReconciler.updateDatastore(test.incomingPod, test.datastore.inferencePool)
-			var gotPods []string
-			test.datastore.pods.Range(func(k, v any) bool {
-				pod := k.(Pod)
+			// Set up the scheme.
+			scheme := runtime.NewScheme()
+			_ = clientgoscheme.AddToScheme(scheme)
+			initialObjects := []client.Object{}
+			if test.incomingPod != nil {
+				initialObjects = append(initialObjects, test.incomingPod)
+			}
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(initialObjects...).
+				Build()
+
+			podReconciler := &PodReconciler{Client: fakeClient, Datastore: test.datastore}
+			namespacedName := types.NamespacedName{Name: test.incomingPod.Name, Namespace: test.incomingPod.Namespace}
+			if test.req == nil {
+				test.req = &ctrl.Request{NamespacedName: namespacedName}
+			}
+			if _, err := podReconciler.Reconcile(context.Background(), *test.req); err != nil {
```

Comment:
Nice! we are testing the Reconcile method instead of testing a private helper method.

Reply:
yeah, we should do that for all controllers.